client/src/js/services/FindEntityService.js

```diff
@@ -0,0 +1,24 @@
+angular.module('bhima.services')
+.service('FindEntityService', FindEntityService);
+
+FindEntityService.$inject = ['$uibModal'];
+
+function FindEntityService(Modal) {
+  var service = this;
+
+  service.openModal = openModal;
+
+  function openModal(row) {
```

Comment:
Passing in the row doesn't make a lot of sense here.  A better API would be to simply return the `ModalInstance` and do the data binding in the controller.  Like this:
```js

// ... in the service ...
function openModal() {
  return Modal.open(params).result;
}

// ... in the controller ...
var instance = FindEntityService.openModal();
instance.then(function (entity) {
  row.entity = entity;
});
```

This provides a nice separation of logic - the service only needs to know how to open a modal, and the controller only needs to know how to use the result of the modal.